src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,447 @@
+"""
+Utilities to manipulate and analyze raster data.
+
+This module extends xarray and rioxarray to provide
+additional functionality for raster processing and
+post-processing. The extension is based on the
+guidance for how to extend xarray:
+
+http://xarray.pydata.org/en/stable/internals/extending-xarray.html
+
+However, since rioxarray already extends xarray, this
+modules extensions inherit from the RasterArray and
+RasterDataset extensions respectively. This ensures
+cleaner code in the module as ``rio`` methods are
+available immediately, but also means a couple of
+design decisions are followed.
+
+The xarray.DataArray and xarray.DataSet
+extensions here inherit from rioxarray base classes.
+Thus, methods that are identical for both objects
+are defined in a mixin class ``AatRasterMixin`` which
+can be inherited by the two respective extensions.
+"""
+
+import logging
+from typing import Any, Callable, Dict, List, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas
+import pandas as pd
+import rioxarray
+import xarray
+import xarray as xr
+from rioxarray.raster_array import RasterArray
+from rioxarray.rioxarray import _get_data_var_message
+
+logger = logging.getLogger(__name__)
+
+
+class AatRasterMixin:
+    """AA toolbox mixin base class."""
+
+    # setting attributes to avoid mypy error, from SO
+    # https://stackoverflow.com/questions/53120262/mypy-how-to-
+    # ignore-missing-attribute-errors-in-mixins/53228204#53228204
+    _get_obj: Callable
+
+    def __init__(self, xarray_obj):
+        super().__init__(xarray_obj)
+
+        # Adding lat/lon to set of default spatial dims
+        if "lat" in self._obj.dims and "lon" in self._obj.dims:
+            self._x_dim = "lon"
+            self._y_dim = "lat"
+
+        # Managing time coordinate default dims
+        self._t_dim = None
+        for t in ["t", "T"]:
+            if t in self._obj.dims:
+                self._t_dim = t
```

Comment:
would "time" aslo be a default? Or is that already added by standard (rio)xarray

Reply:
hmm I see later on the `set_time_dim()` where you use `time` as an example. Maybe I am missing the logic, but why is `time` not in the standard set (as I have seen that being used quite often)?

Reply:
There is definitely no good logic to it and happy to increase the default set as necessary! Was just putting in what I'd commonly seen. Have now added `time` and if you have any other defaults you think are good, let me know @Tinkaa @turnerm. I hadn't seen `time` before but had also seen `F` out there, but thought that might not be an appropriate default and we should just require manually specifying in the `datasource`?

Reply:
Yes definitely wouldn't add `F`. Only have seen that used by IRI related to forecasted times. But saw you added `time` which seems good